Moi hcu 011 nested form

leninkp opened the pull request.

leninkp committed:
created model content

macool commented:
needs rebasing

macool commented on Gemfile:
```diff
@@ -66,6 +66,21 @@ group :staging do
+gem 'spellchecker'
+
+#form
+gem "nested_form", :git => "git://github.com/fabricioflores/nested_form.git"
```
a excepción de esta línea, todas las demás tienen que ser quitadas. estas gemas ya están

macool commented on app/controllers/admin/neurons_controller.rb:
```diff
@@ -39,6 +39,13 @@ def index
+    def preview
+      respond_to do |format|
+        format.js
+      end
```
:question:

macool commented on app/helpers/content_helper.rb:
```diff
@@ -0,0 +1,25 @@
+      end
+    end
+  end
+end
```
me parece que esto va en decoradores

macool commented on app/helpers/content_helper.rb:
```diff
@@ -0,0 +1,25 @@
+
+  def content_kind
+    I18n.t("views.contents.kinds").map { |key, value| [ value, key ] }
+  end
```
no es mejor tener estos en el controlador? ^

leninkp replied:
Por qué en el controlador?

macool replied:
es información que va a la DB. Ahora que lo pienso tampoco puede depender del idioma

macool replied:
tenemos que pensar mejor en cómo manejamos estos atributos. https://github.com/GrowMoi/moi/pull/33/files#diff-526f3087b6dbbf77083d177614d162f6R57

macool commented on app/models/content.rb:
```diff
@@ -0,0 +1,20 @@
+  belongs_to :neuron
+
+  #validations
+  validates :description, :presence =>true
```
preferimos la nueva hash syntax
`presence: true`

macool commented on app/models/neuron.rb:
```diff
@@ -10,9 +10,15 @@
   has_paper_trail only: [:title, :parent_id]
 
+  #nested
+  accepts_nested_attributes_for :contents, :allow_destroy => true, reject_if: proc { |attributes| attributes['description'].blank? }
```
esta línea es muy larga

macool commented on app/views/admin/neurons/_content_form.html.slim:
```diff
@@ -0,0 +1,16 @@
+.form-group
+  = content.label :description
+  = content.text_area :description, class: "form-control"
+  = content.hidden_field :neuron_id
```
no creo que necesitemos este `id`

macool commented on app/views/admin/neurons/_content_form.html.slim:
```diff
@@ -0,0 +1,16 @@
+  = content.text_area :description, class: "form-control"
+  = content.hidden_field :neuron_id
+
+= content.link_to_remove t("helpers.custom_buttons.remove_content"), class: "btn btn-danger"
```
esto no es `custom_buttons`. podría ser `neuron_contents`

macool commented on app/views/admin/neurons/_form.html.slim:
```diff
@@ -1,4 +1,4 @@
-= form_for [:admin, neuron] do |f|
+= nested_form_for [:admin, neuron], :html => {:class => "neuron_form"} do |f|
```
para qué necesitamos la clase? ^

leninkp replied:
es error mio, lo necesitaba en el preview

macool commented on app/views/admin/neurons/_form.html.slim:
```diff
@@ -8,15 +8,16 @@
-    - unless neuron.versions.empty?
-      = link_to t("views.neurons.show_changelog"),
-                log_admin_neuron_path(neuron),
-                class: "btn btn-link"
```
tal vez esto se perdió en el rebase ^

macool commented on config/routes.rb:
```diff
@@ -60,9 +60,10 @@
         get "log"
+        get 'contents'
       end
+      resources :contents
```
para qué tenemos esto?

leninkp replied:
son cambios anteriores, lo voy a eliminar

macool commented on app/assets/javascripts/admin/moi_tree.js.coffee:
```diff
@@ -135,6 +135,8 @@ class MoiTree
     $popover.find(".edit-link")
             .attr("href", "/admin/neurons/#{node.id}/edit")
+    $popover.find(".content-link")
+            .attr("href", "/admin/neurons/#{node.id}/contents")
```
y este link?

leninkp replied:
es de cambios anteriores, lo voy a borrar

macool committed:
[MOI-HCU-011] nested contents form enhances

enhance behaviour for neuron's contents

macool committed:
minor fixes. closes #36

fabricioflores committed:
Merge pull request #35 from GrowMoi/moi-hcu-011-macool-enhances

[MOI-HCU-011] nested contents form enhances

macool committed:
use chosen

macool committed:
update specs

as chosen is applied to all selects, they are hidden. so capybara selector should now specify `visible: false`

fabricioflores committed:
Merge pull request #37 from GrowMoi/chosen

use chosen

leninkp commented on config/locales/es.yml:
```diff
@@ -34,11 +34,31 @@
+      updated: "Contenido actualizado"
+      show: "Mostrar contenidos"
+      levels:
+        "1": "Principiante"
```
Esta bien esta parte?, no debería ser por ejm:  `1` en ves de `"1"` @macool, es decir sin comillas.

macool replied:
no. busca strings

leninkp committed:
amends

macool commented on app/assets/javascripts/admin/default_tabs.js.coffee:
```diff
@@ -0,0 +1,21 @@
+  # show first default list group by default
+  $(".default-list-group").each ->
+    $(this).find("a:first").addClass("active")
+                           .tab "show"
```
mover esto a `default_list_group.js`

leninkp replied:
entendido

macool commented:
se ve bien. Nos falta cambiar el show para que se vea, igual que en el formulario, organizado en tabs

macool commented:
LGTM
